File: AspNetMvcUnleashed/0672329980_source/Chapter11/VB/MvcApplication1/MvcApplication1/Controllers/MovieController.vb
Public Class MovieController
    Inherits Controller

    Private _entities As New MoviesDBEntities()

    <OutputCache(Duration:=9999, VaryByParam:="movieId")> _
    Public Function Details(ByVal movieId As Integer) As ActionResult
        Dim result = (From movie In _entities.MovieSet _
                      Where movie.Id = movieId _
                      Select movie).FirstOrDefault()
        Return View(result)
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter11/CS/MvcApplication1/MvcApplication1/Controllers/MovieController.cs
using System.Linq;
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class MovieController : Controller
    {
        private MoviesDBEntities _entities = new MoviesDBEntities(); 

        [OutputCache(Duration=9999, VaryByParam="movieId")]
        public ActionResult Details(int movieId)
        {
            var result = (from movie in _entities.MovieSet
                         where movie.Id == movieId
                         select movie).FirstOrDefault();
            return View(result);
        }

    }
}
